src/js/modules/infragistics.ui.editors.js

```diff
@@ -10750,12 +10720,21 @@
 					} else {
 						date = self._setNewDateMidnight();
 					}
-					date = self._setDateField("year", date, dateFromPicker.getFullYear());
 
-					//Temporary change the date to be in the middle of the month 15th, because when using JavaScript Date object to set month when date is 31, the date object is moved with one day.
-					date = self._setDateField("date", date, 15);
-					date = self._setDateField("month", date, dateFromPicker.getMonth());
-					date = self._setDateField("date", date, dateFromPicker.getDate());
+					if (self.options.displayTimeOffset !== null) {
```

Comment:
I checked the coverage and the date picker is tested only when `displayTimeOffset` is default (`null`). Please add date picker tests with different values for the `displayTimeOffset`